scm-step/src/main/java/org/jenkinsci/plugins/workflow/steps/scm/GitStep.java

```diff
@@ -56,21 +79,39 @@ public String getBranch() {
         return branch;
     }
 
+    public String getCredentialsId() {
+        return credentialsId;
+    }
+
     @DataBoundSetter public void setBranch(String branch) {
         this.branch = branch;
     }
 
+    @DataBoundSetter public void setCredentialsId(String credentialsId) {
+        StandardCredentials credentials = lookupCredentials(null, credentialsId, url);
+        if (credentials == null) throw new IllegalArgumentException("invalid Credentials Id "+credentialsId);
```

Comment:
You cannot do this check here because you do not have the `Item context` which you need to look up folder credentials. Just skip it; if the credentials are missing, this will get reported by `GitSCM` anyway.